builtin/providers/triton/resource_machine.go

```diff
@@ -0,0 +1,436 @@
+				Optional:    true,
+				Default:     false,
+			},
+
+			// computed resources from metadata
+			"root_authorized_keys": {
+				Description: "authorized keys for the root user on this machine",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"user_script": {
+				Description: "user script to run on boot (every boot on SmartMachines)",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"user_data": {
+				Description: "copied to machine on boot",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"administrator_pw": {
+				Description: "administrator's initial password (Windows only)",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceMachineCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudapi.Client)
+
+	var networks []string
+	for _, network := range d.Get("networks").([]interface{}) {
+		networks = append(networks, network.(string))
+	}
+
+	metadata := map[string]string{}
+	for schemaName, metadataKey := range resourceMachineMetadataKeys {
+		if v, ok := d.GetOk(schemaName); ok {
+			metadata[metadataKey] = v.(string)
+		}
+	}
+
+	tags := map[string]string{}
+	for k, v := range d.Get("tags").(map[string]interface{}) {
+		tags[k] = v.(string)
+	}
+
+	machine, err := client.CreateMachine(cloudapi.CreateMachineOpts{
+		Name:            d.Get("name").(string),
+		Package:         d.Get("package").(string),
+		Image:           d.Get("image").(string),
+		Networks:        networks,
+		Metadata:        metadata,
+		Tags:            tags,
+		FirewallEnabled: d.Get("firewall_enabled").(bool),
+	})
+	if err != nil {
+		return err
+	}
+
+	err = waitForMachineState(client, machine.Id, machineStateRunning, machineStateChangeTimeout)
+	if err != nil {
+		return err
+	}
+
+	// refresh state after it provisions
+	d.SetId(machine.Id)
+	err = resourceMachineRead(d, meta)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceMachineExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	client := meta.(*cloudapi.Client)
+
+	machine, err := client.GetMachine(d.Id())
+
+	return machine != nil && err == nil, err
+}
+
+func resourceMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudapi.Client)
+
+	machine, err := client.GetMachine(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.SetId(machine.Id)
+	d.Set("name", machine.Name)
+	d.Set("type", machine.Type)
+	d.Set("state", machine.State)
+	d.Set("dataset", machine.Dataset)
+	d.Set("memory", machine.Memory)
+	d.Set("disk", machine.Disk)
+	d.Set("ips", machine.IPs)
+	d.Set("tags", machine.Tags)
+	d.Set("created", machine.Created)
+	d.Set("updated", machine.Updated)
+	d.Set("package", machine.Package)
+	d.Set("image", machine.Image)
+	d.Set("primaryip", machine.PrimaryIP)
+	d.Set("networks", machine.Networks)
```

Comment:
Probably worth checking the error return here since the value is not primitive, particularly given the comment above on the schema.